imessage.py

```diff
@@ -80,148 +77,238 @@ class Attachment:
 
 @dataclass
-class iMessage:
-    """Represents an iMessage"""
+class SMSReflectedMessage(Message):
+    @staticmethod
+    def from_raw(message: bytes, sender: str | None = None) -> "SMSReflectedMessage":
+        """Create a `SMSReflectedMessage` from raw message bytes"""
 
-    text: str = ""
-    """Plain text of message, always required, may be an empty string"""
-    xml: str | None = None
-    """XML portion of message, may be None"""
-    participants: list[str] = field(default_factory=list)
-    """List of participants in the message, including the sender"""
-    sender: str | None = None
-    """Sender of the message"""
-    id: uuid.UUID | None = None
-    """ID of the message, will be randomly generated if not provided"""
-    group_id: uuid.UUID | None = None
-    """Group ID of the message, will be randomly generated if not provided"""
-    body: BalloonBody | None = None
-    """BalloonBody, may be None"""
-    effect: str | None = None
-    """iMessage effect sent with this message, may be None"""
+        # Decompress the message
+        try:
+            message = gzip.decompress(message)
+            compressed = True
+        except:
+            compressed = False
 
-    _compressed: bool = True
-    """Internal property representing whether the message should be compressed"""
+        message = plistlib.loads(message)
 
-    _raw: dict | None = None
-    """Internal property representing the original raw message, may be None"""
+        logger.info(f"Decoding SMSReflectedMessage: {message}")
 
-    def attachments(self) -> list[Attachment]:
-        if self.xml is not None:
-            return [Attachment(self._raw, elem) for elem in ElementTree.fromstring(self.xml)[0] if elem.tag == "FILE"]
-        else:
-            return []
-
-    def sanity_check(self):
-        """Corrects any missing fields"""
-        if self.id is None:
-            self.id = uuid.uuid4()
-
-        if self.group_id is None:
-            self.group_id = uuid.uuid4()
-
-        if self.sender is None:
-            if len(self.participants) > 1:
-                self.sender = self.participants[-1]
-            else:
-                logger.warning(
-                    "Message has no sender, and only one participant, sanity check failed"
-                )
-                return False
+        return SMSReflectedMessage(
+            text=message["mD"]["plain-body"],  # type: ignore
+            sender=sender, # type: ignore
+            participants=[re["id"] for re in message["re"]] + [sender], # type: ignore
+            id=uuid.UUID(message["mD"]["guid"]), # type: ignore
+            _raw=message, # type: ignore
+            _compressed=compressed,
+        )
 
-        if self.sender not in self.participants:
-            self.participants.append(self.sender)
+    def to_raw(self) -> bytes:
+        #  {'re': [{'id': '+14155086773', 'uID': '4155086773', 'n': 'us'}], 'ic': 0, 'mD': {'handle': '+14155086773', 'guid': imessage.py:201
```

Comment:
is this a comment you want to leave in? (since it has a private phone number)

Reply:
that's not actually a private phone number, iirc it was for some chatgpt-over-sms service? or some other test number.